### PR TITLE
XFS support

### PR DESCRIPTION
**Pull request by vpodzime**
We already have support for the *ext4* file system. *xfs* is the other one that is most commonly used, so it should be supported too.

### CODE REVIEW

File: src/lib/plugin_apis/fs.api [hunk 1]
```diff
@@ -147,4 +196,77 @@ BDFSExt4Info* bd_fs_ext4_get_info (gchar *device, GError **error);
...
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an xfs signature was successfully wiped from the @device or
+ *          not
+ */
+gboolean bd_fs_xfs_wipe (gchar *device, GError **error);
+
+/**
+ * bd_fs_xfs_check:
+ * @device: the device the file system on which to check
```
Comment (dwlehman) [inline src/lib/plugin_apis/fs.api]: Something like "the device containing the file system to check" would be clearer.

File: src/lib/plugin_apis/fs.api [hunk 2]
```diff
@@ -147,4 +196,77 @@ BDFSExt4Info* bd_fs_ext4_get_info (gchar *device, GError **error);
...
+ * Returns: whether an xfs file system on the @device is clean or not
+ *
+ * Note: if the file system is mounted it may be reported as unclean even if
+ *       everything is okay and there are just some pending/in-progress writes
+ */
+gboolean bd_fs_xfs_check (gchar *device, GError **error);
+
+/**
+ * bd_fs_xfs_repair:
+ * @device: the device the file system on which to repair
```
Comment (dwlehman) [inline src/lib/plugin_apis/fs.api]: same here as above ("the device containing the file system to repair")

**Comment by dwlehman** [comment]
Aside from the wording of several comments (all along the same lines in the same file) this looks good to me.

**Comment by vpodzime** [comment]
Fixed the wording. I'm going to wait for the check to run and finish and then merge this if everything looks okay.

**Pull request merged**